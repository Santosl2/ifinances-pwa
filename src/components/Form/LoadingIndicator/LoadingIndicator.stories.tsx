import { ComponentMeta, ComponentStory } from "@storybook/react";

import { LoadingIndicator } from "./LoadingIndicator";

export default {
  title: "Components/LoadingIndicator",
  component: LoadingIndicator,
} as ComponentMeta<typeof LoadingIndicator>;

const Template: ComponentStory<typeof LoadingIndicator> = (args) => (
  <LoadingIndicator {...args} />
);

export const Default = Template.bind({});

Default.args = {
  color: "#bbbbbb",
  width: "50px",
  height: "50px",
};
