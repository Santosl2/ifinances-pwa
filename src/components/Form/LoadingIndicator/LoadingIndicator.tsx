import styled, { keyframes } from "styled-components";

const spinner = keyframes`
    from {
        transform: rotate(0);
    }
    to {
        transform: rotate(360deg);
    }
`;

export const LoadingIndicator = styled.div`
  width: 20px;
  height: 20px;
  background: transparent;
  padding: 5px;
  border: 2px solid transparent;
  border-right-color: var(--loading-border-color);
  margin: 0 auto;
  border-radius: 1rem;
  animation: ${spinner} 1.5s infinite;
`;
