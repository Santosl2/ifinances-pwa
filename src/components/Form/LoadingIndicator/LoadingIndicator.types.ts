export interface LoadingIndicatorStylesProps {
  color?: string;
  height?: string;
  width?: string;
}
