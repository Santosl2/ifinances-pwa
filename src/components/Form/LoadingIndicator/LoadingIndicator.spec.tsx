/* eslint-disable testing-library/no-node-access */
/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable unused-imports/no-unused-imports */
import * as React from "react";

import { render, screen } from "@testing-library/react";

import { LoadingIndicator } from "./LoadingIndicator";

describe("Loading Test", () => {
  it("should render Loading", () => {
    const { container } = render(<LoadingIndicator />);

    expect(container.firstChild).toHaveStyle({
      borderRadius: "1rem",
      background: "transparent",
    });
  });
});
