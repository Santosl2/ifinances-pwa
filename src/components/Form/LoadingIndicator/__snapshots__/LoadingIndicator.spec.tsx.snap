// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Test should render Loading 1`] = `
.c0 {
  width: 20px;
  height: 20px;
  background: transparent;
  padding: 5px;
  border: 2px solid transparent;
  border-right-color: var(--loading-border-color);
  margin: 0 auto;
  border-radius: 100%;
  -webkit-animation: bmMtkP 1.5s infinite;
  animation: bmMtkP 1.5s infinite;
}

<div
  class="c0"
/>
`;
